src/Trains.NET.Rendering/Game/Game.cs

```diff
@@ -25,38 +21,42 @@ public class Game : IGame
         private readonly IImageFactory _imageFactory;
         private readonly PerSecondTimedStat _skiaFps = InstrumentationBag.Add<PerSecondTimedStat>("Draw-FPS-Skia");
         private readonly ElapsedMillisecondsTimedStat _skiaDrawTime = InstrumentationBag.Add<ElapsedMillisecondsTimedStat>("Draw-Skia-AllUp");
-        private readonly ElapsedMillisecondsTimedStat _gameBufferReset = InstrumentationBag.Add<ElapsedMillisecondsTimedStat>("Draw-Game-BufferReset");
         private readonly Dictionary<ILayerRenderer, ElapsedMillisecondsTimedStat> _renderLayerDrawTimes;
         private readonly Dictionary<IScreen, ElapsedMillisecondsTimedStat> _screenDrawTimes;
         private readonly Dictionary<ILayerRenderer, ElapsedMillisecondsTimedStat> _renderCacheDrawTimes;
-        private readonly Dictionary<ILayerRenderer, IImage> _imageBuffer = new();
         private readonly ITimer _renderLoop;
         private readonly IEnumerable<IScreen> _screens;
-        private readonly Dictionary<IScreen, IImage> _screenCache = new Dictionary<IScreen, IImage>();
+        private readonly IImageCache _imageCache;
 
         public Game(IGameBoard gameBoard,
                     IEnumerable<ILayerRenderer> boardRenderers,
                     IPixelMapper pixelMapper,
                     IImageFactory imageFactory,
                     ITimer renderLoop,
-                    IEnumerable<IScreen> screens)
+                    IEnumerable<IScreen> screens,
+                    IImageCache imageCache)
         {
             _gameBoard = gameBoard;
             _boardRenderers = boardRenderers;
             _pixelMapper = pixelMapper;
             _imageFactory = imageFactory;
             _renderLoop = renderLoop;
             _screens = screens;
+            _imageCache = imageCache;
 
-            foreach (var screen in _screens)
+            foreach (IScreen screen in _screens)
+            {
+                screen.Changed += (s, e) => _imageCache.SetDirty(screen);
+            }
+            foreach (ICachableLayerRenderer renderer in _boardRenderers.OfType<ICachableLayerRenderer>())
             {
-                screen.Changed += (s, e) => _screenCache.Remove(screen);
+                renderer.Changed += (s, e) => _imageCache.SetDirty(renderer);
             }
 
             _renderLayerDrawTimes = _boardRenderers.ToDictionary(x => x, x => InstrumentationBag.Add<ElapsedMillisecondsTimedStat>(GetLayerDiagnosticsName(x)));
             _screenDrawTimes = _screens.ToDictionary(x => x, x => InstrumentationBag.Add<ElapsedMillisecondsTimedStat>(GetLayerDiagnosticsName(x)));
             _renderCacheDrawTimes = _boardRenderers.Where(x => x is ICachableLayerRenderer).ToDictionary(x => x, x => InstrumentationBag.Add<ElapsedMillisecondsTimedStat>("Draw-Cache-" + x.Name.Replace(" ", "")));
-            _pixelMapper.ViewPortChanged += (s, e) => _needsBufferReset = true;
+            _pixelMapper.ViewPortChanged += (s, e) => _imageCache.SetDirtyAll(_boardRenderers);
```

Comment:
Now that setting buffers to dirty doesn't mean they need to be redrawn, we don't need the bool flag to synchronize clearing the cache